include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -185,26 +185,50 @@ using __kernel_name_generator =
     _BaseName<_CustomName>;
 #endif
 
-template <typename _DerivedKernelName>
+template <typename... _KernelNames>
 class __kernel_compiler
 {
+    static constexpr int __n_kernel = sizeof...(_KernelNames);
+    using _KernelArray = std::array<sycl::kernel, __n_kernel>;
```

Comment:
Do we still use the rule for `::std::` in product code? if so, please fix that everywhere